internal/stacks/stack.go

```diff
@@ -0,0 +1,80 @@
+package stacks
+
+import (
+	"emperror.dev/errors"
+	"encoding/json"
+	"github.com/aviator-co/av/internal/git"
+	"strings"
+)
+
+type Tree struct {
+	Previous *Tree
+	Next     []*Tree
+	Branch   *BranchMetadata
+}
+
+type GetTreeOpts struct {
+	Root string
+}
+
+func GetTrees(repo *git.Repo) (map[string]*Tree, error) {
+	refs, err := repo.ListRefs(&git.ListRefs{
+		Patterns: []string{"refs/av/stack-metadata/*"},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	refNames := make([]string, len(refs))
+	for i, ref := range refs {
+		refNames[i] = ref.Name
+	}
+	refContents, err := repo.CatFileBatch(&git.CatFileBatch{
+		Revisions: refNames,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	metas := make(map[string]*BranchMetadata)
+	children := make(map[string][]string)
+	hasParent := make(map[string]bool)
+	for _, ref := range refContents {
+		var meta BranchMetadata
+		meta.Name = strings.TrimPrefix(ref.Revision, "refs/av/stack-metadata/")
+		if err := json.Unmarshal(ref.Contents, &meta); err != nil {
+			return nil, errors.WrapIff(err, "failed to unmarshal metadata for branch %q", ref.Revision)
+		}
+		metas[meta.Name] = &meta
+		hasParent[meta.Name] = true
+		// set default value for parent
+		hasParent[meta.Parent] = hasParent[meta.Parent]
+		children[meta.Parent] = append(children[meta.Parent], meta.Name)
```

Comment:
something is weird about this whole structure. Are we capturing the scenario where reference has multiple children?